### PR TITLE
Add multi file upload support

### PR DESCRIPTION
**Pull request by anan0120**
# What changes were proposed in this pull request?
- [ ] Add multi file upload support

# How was this patch tested?
- [ ] Testing done

# Are there and DOC need to update?
- [ ] Doc is finished

# Spark Core Compatibility

### CODE REVIEW

**Comment by chncaesar** [comment]
Issue #1737

File: streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala [hunk 1]
```diff
@@ -371,26 +371,42 @@ class MLSQLRest(override val uid: String) extends MLSQLSource
...
+          entity.addBinaryBody(fileInfo._2, inputStream, ContentType.MULTIPART_FORM_DATA, fileInfo._2)
+        })
 
         params.filter(_._1.startsWith("form.")).
           filter(v => v._1 != "form.file-path" && v._1 != "form.file-name").foreach { case (k, v) =>
           entity.addTextBody(k.stripPrefix("form."), Templates2.dynamicEvaluateExpression(v, ScriptSQLExec.context().execListener.env().toMap))
         }
-        request.body(entity.build()).execute()
-      case (_, v) =>
+        request.body(entity.build()).execute()      case (_, v) =>
```
Comment (chncaesar) [inline streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala]: case (_, v) => 需要换行

File: streamingpro-mlsql/src/test/scala/streaming/test/Test.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ object Test {
         .add("sql",
           """
             | set data='''
-            |{"key":"yes","value":"no","topic":"test","partition":0,"offset":0,"timestamp":"2008-01-24 18:01:01.001","timestampType":0}
```
Comment (chncaesar) [inline streamingpro-mlsql/src/test/scala/streaming/test/Test.scala]: Please add unit test to `MLSQLRestTest`. It contains several cases. Please take one as an example.

**Review by chncaesar — CHANGES_REQUESTED** [review]

**Comment by anan0120** [comment]
1、case (_, v) => 需要换行    已处理
2、无法进行mock，程序中有实际路径的判断

File: streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala [hunk 3]
```diff
@@ -371,26 +371,42 @@ class MLSQLRest(override val uid: String) extends MLSQLSource
         request.bodyForm(form.build(), Charset.forName("utf-8")).execute()
 
       case ("post", contentType) if contentType.trim.startsWith("multipart/form-data") =>
+        request.removeHeaders("Content-Type") // Avoid setting boundary
```
Comment (hellozepp) [inline streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala]: 请不要使用行尾注释，添加到该行的上面

**Comment by hellozepp** [comment]
@anan0120 你可以添加一个测试脚本用例到 `byzer-lang/streamingpro-it/src/test/resources/sql/simple` 下面吗，可以参考例子09_syntax_analysis_ext_test.mlsql

File: streamingpro-mlsql/src/test/scala/streaming/test/Test.scala [hunk 4]
```diff
@@ -36,7 +36,7 @@ object Test {
...
+            |{"key":"yesvalue","value":"no","topic":"test","partition":0,"offset":0,"timestamp":"2008-01-24 18:01:01.001","timestampType":0}
```
Comment (hellozepp) [inline streamingpro-mlsql/src/test/scala/streaming/test/Test.scala]: 这个修改是必要的吗

**Review by hellozepp — CHANGES_REQUESTED** [review]

File: streamingpro-mlsql/src/test/scala/streaming/test/Test.scala (hunk 4 above)
Comment (anan0120) [reply streamingpro-mlsql/src/test/scala/streaming/test/Test.scala]: 这个确实没必要修改，当时做了临时的一个测试，遗留到了这次PR上

Comment (anan0120) [reply streamingpro-mlsql/src/test/scala/streaming/test/Test.scala]: 已复原

File: streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala (hunk 3 above)
Comment (anan0120) [reply streamingpro-mlsql/src/main/java/tech/mlsql/datasource/impl/MLSQLRest.scala]: 已调整

**Review by chncaesar — APPROVED** [review]

**Review by hellozepp — APPROVED** [review]

**Pull request merged**